migrations/2_deploy_contracts.js

```diff
@@ -42,16 +42,38 @@ async function deploy(options, owner, config) {
 
   console.log('Deploying upgradeDelegator and admin');
   const admin = await ProxyAdmin.new();
+  let upgradeDelegator;
 
-  const upgradeDelegator = await UpgradeDelegator.new();
-  await upgradeDelegator.initialize(governor.address, admin.address);
+  //check if the script if BlockableUpgradeDelegator should be used
```

Comment:
There is a reason why we do not use the upgradeDelegator as a proxy and that is because it is too close to the proxyAdmin and it can cause some problems. Besides that, I think it is better to change the upgradeDelegator if we want to upgrade because it doesn't have any complex storage value that we want to preserve.

Reply:
This is the script for the first deploy of the contracts. The idea is that if the block date is set for a first deploy then use BlockableUpgradeDelegator to save the gas from an Upgrade.
The next step is to write the upgredability scripts to update UpgradeDelegator by BlockableUpgradeDelegator.